script/fetch-issues.js

```diff
@@ -1,64 +1,208 @@
-  process.exit()
-}
+const GH_GQL_BASE = 'https://api.github.com/graphql';
+const GH_GQL_OPTIONS = {
+  url: GH_GQL_BASE,
+  headers: process.env.GITHUB_TOKEN
+    ? { Authorization: `bearer ${process.env.GITHUB_TOKEN}` }
+    : {}
+};
 
-console.log('Fetching issues data for', repo)
+const client = GraphQL(GH_GQL_OPTIONS);
 
-;(async () => {
-  const browser = await puppeteer.launch()
-  const page = await browser.newPage()
+// number of pages we want to query (default: 20 pages x 100 items)
+// if PAGE_THRESHOLD is -1, then fetch all issues/PRs
+const pageThreshold = process.env.PAGE_THRESHOLD || 20;
+console.log('page number threshold:', pageThreshold);
 
-  const refresh = async () =>
-    await page.goto(`http://localhost:8080/#/r/${repo}/kanban`)
+const repo = process.env.REPOSITORIES;
+
+if (!repo) {
+  console.log('No repositories to cache. Skipping.');
+  process.exit();
+}
 
-  await refresh()
-  await page.evaluate(token => {
-    localStorage.setItem('gh-token', token)
-  }, token)
-  await refresh()
+const repoOwner = repo.split(':')[0];
+const repoNames = repo.substring(repo.indexOf(':') + 1).split('|');
 
-  await page.waitForSelector('.kanban-board', { timeout: 5 * 60 * 1000 })
+console.log('Fetching issues data for', repoOwner, repoNames);
 
-  let cache = await page.evaluate(() => {
-    const dump = key => {
-      return new Promise(resolve => {
-        const open = indexedDB.open(key, 1)
+const getTime = timeString => moment(timeString).toDate().getTime();
 
-        open.onsuccess = () => {
-          const db = open.result
-          const tx = db.transaction(key)
-          const store = tx.objectStore(key)
+async function fetchNextPage(owner, name, isIssue, data, cursor, pageCount) {
+  // fetch data of next page
+  console.log('owner:', owner, 'name:', name, 'isIssue:', isIssue,
+    'cursor:', cursor, 'page count:', pageCount);
 
-          store.getAll().onsuccess = event => {
-            resolve(event.target.result)
-          }
+  let nextData = null, hasPreviousPage = false;
+  pageCount++;
+  try {
+    if (isIssue) {
+      ({ data: nextData } = await client.query(
+        GITHUB_ISSUE_INFO_QUERY,
+        {owner, name, before: cursor}
+      ));
+    }
+    else {
+      ({ data: nextData } = await client.query(
+        GITHUB_PR_INFO_QUERY,
+        {owner, name, before: cursor}
+      ));
+    }
+  } catch (error) {
+    console.log('owner:', owner, 'name:', name, 'isIssue:', isIssue,
+      'cursor:', cursor, 'error:', error);
+  }
+  if (nextData && nextData.repository) {
+    let nodes = [];
+    if (isIssue) {
+      ({ nodes, pageInfo } = nextData.repository.issues);
+    }
+    else {
+      ({ nodes, pageInfo } = nextData.repository.pullRequests);
+    }
+    cursor = pageInfo.startCursor;
+    hasPreviousPage = pageInfo.hasPreviousPage;
+    const result = nodes.map(node => {
+      let info = {
+        repoOwner: owner,
+        repoName: name,
+        updatedAtMs: getTime(node.updatedAt),
```

Comment:
this should be `updatedAt` ; that is what `database.js` looks for.

Reply:
oh ... that is at the repo level .. hmm.. this might be ok...
```
git grep updatedAtMs
script/fetch-issues.js:        updatedAtMs: moment(node.updatedAt).toDate().getTime(),
src/database.js:      updatedAtMs: moment(issue.updatedAt).toDate().getTime(),
```